### PR TITLE
Scarab Salvage Ship

### PR DESCRIPTION
**Pull request by RustingWithYou**
Adds a new away site which can spawn in the Coalition, a Scarab vessel salvaging and harvesting resources for the main fleet.

### CODE REVIEW

**Comment by RustingWithYou** [comment]
!review

**Comment by DreamySkrell** [comment]
To be 100% honest, I do not like this design.

Having to go EVA to set up or refuel propulsion, or to go between the shuttles, is a massive pain in the ass with how slow airlocks sometimes get, especially when the server is lagging. Standing in airlocks and waiting for them to cycle, is not very conductive to roleplay. There's also no external docks here either, so anyone visiting has to go EVA as well.

I would expect that scarabs, having spent all that time in space, would be much more efficient about ship design, preferring instead a small and tight ship where everything is crowded and close, and does not waste any space.

**Comment by DreamySkrell** [comment]
image
These airlocks need more than just one shared airlock air canister. Make it at least 3, and connect it to the air loop with a pump so they can get refilled (there's an example of this on top deck horizon, airlock near tcomms).

---

image
image
This landmark is not compliant with the guidelines, does not have enough space on the sides. I would recommend to use the ruler helpers like I do on my (unfinished) SCC scout ship - they don't spawn actually anything in game, so they're map editor helpers only. <https://github.com/Aurorastation/Aurora.3/pull/17826/checks?check_run_id=18976895344>

---

image
Since this ship has no docking ports, it should at the very least have these catwalk "docks" instead, so anyone visiting just has to walk on catwalks, and not have to fly with jetpacks or whatever. See my SCC scout ship for inspiration, I think that one is very nice about this, in that it has both docking ports, and these catwalk "docks".

---

image
These catwalks connecting the shuttles should be 2-tiles wide at least, so they're not super annoying when moving cargo or whatever around.

---

image
This airlock is not complaint with the guidelines - there can't be anything protruding in front of the airlock "face". It couldn't dock to anything.

---

image
Do they live on this ship? I would expect the crew to have some like, personal space for themselves. It doesn't have to be that every crew member has their private room, but like, a table at least would be nice. As it is currently, the bunk beds just look like a last-minute addition just to tick the "crew beds" checkbox. Especially so in how they're sharing room with the flak ship gun...

**Comment by DreamySkrell** [comment]
image
image
These two airlocks do not have buttons on the interior side.

image
None of the airlocks on this map have airlock sensors. Do they actually work in game without them? I heavily doubt they do. Please test your maps before putting them up for review.

**Review by DreamySkrell — CHANGES_REQUESTED** [review]
See previous comments

**Comment by RustingWithYou** [comment]
> See previous comments

Most of this has been addressed. Catwalks are wider with more docks, airlocks have been fixed, and the bedroom and armory have been repositioned so that the space is a bit more convenient.

**Review by NonQueueingMatt — APPROVED** [review]

**Comment by DreamySkrell** [comment]
please ping me on discord when this is ready for review again

File: maps/away/ships/coc/coc_scarab/coc_scarab.dm [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+		"scarab_nav3",
+		"scarab_nav4",
+		"scarab_dock1",
+		"scarab_dock2",
+		"scarab_dock3",
+		"scarab_dock4",
+		"scarab_dock5"
+	)
+	initial_restricted_waypoints = list(
+		"Intrepid" = list("scarab_intrepid"),
```
Comment (DreamySkrell) [inline maps/away/ships/coc/coc_scarab/coc_scarab.dm:48]: no docks restricted to shuttles - this should just be a normal dock that any shuttle can use (exceptions of course being their hangars or starting docks, or events, or etc)

File: maps/away/ships/coc/coc_scarab/coc_scarab.dm [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+/obj/effect/shuttle_landmark/coc_scarab/dock4
+	name = "Scarab Salvage Vessel - Starboard Dock #2"
+	landmark_tag = "scarab_dock4"
+
+/obj/effect/shuttle_landmark/coc_scarab/dock5
+	name = "Scarab Salvage Vessel - Starboard Dock #3"
+	landmark_tag = "scarab_dock5"
+
+/obj/effect/shuttle_landmark/coc_scarab/intrepid
```
Comment (DreamySkrell) [inline maps/away/ships/coc/coc_scarab/coc_scarab.dm:97]: same as previous comment

File: maps/away/ships/coc/coc_scarab/coc_scarab_areas.dm [hunk 3]
```diff
@@ -0,0 +1,79 @@
+/area/coc_scarab
+	name = "Scarab Salvage Vessel"
+	icon_state = "blue"
+	requires_power = 1
```
Comment (DreamySkrell) [inline maps/away/ships/coc/coc_scarab/coc_scarab_areas.dm:4]: TRUE instead of 1

File: code/modules/overmap/ships/computers/helm.dm [hunk 4]
```diff
@@ -27,7 +27,7 @@
 /obj/machinery/computer/ship/helm/terminal
 	name = "helm control terminal"
 	icon = 'icons/obj/machinery/modular_terminal.dmi'
-	icon_screen = "helm"
+	icon_screen = "nav"
```
Comment (DreamySkrell) [inline code/modules/overmap/ships/computers/helm.dm:30]: why?

Comment (RustingWithYou) [reply code/modules/overmap/ships/computers/helm.dm:30]: i actually have no idea why this is in here

**Review by DreamySkrell — APPROVED** [review]

**Pull request merged**